src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs

```diff
@@ -457,5 +457,138 @@ private async Task InnerApplySchemaTransformersAsync(IOpenApiSchema inputSchema,
     }
 
     private JsonNode CreateSchema(OpenApiSchemaKey key)
-        => JsonSchemaExporter.GetJsonSchemaAsNode(_jsonSerializerOptions, key.Type, _configuration);
+    {
+        var schema = JsonSchemaExporter.GetJsonSchemaAsNode(_jsonSerializerOptions, key.Type, _configuration);
+        return ResolveReferences(schema, schema);
+    }
+
+    private static JsonNode ResolveReferences(JsonNode node, JsonNode rootSchema)
+    {
+        return ResolveReferencesRecursive(node, rootSchema, []);
+    }
+
+    private static JsonNode ResolveReferencesRecursive(JsonNode node, JsonNode rootSchema, HashSet<string> visitedRefs)
```

Comment:
Does this need something to stop an infinite loop happening? IIRC, there's other bits of the code that do recursion that do something like _"this is the 64th time I'm calling this, so that's too much - stop"_.

Reply:
For handling recursion, we rely on things being caught by JsonSchemaExporter before the schema is returned to us. Tests like `ThrowsForOverlyNestedSchemas` and `SupportsDeeplyNestedSchemaWithConfiguredMaxDepth` cover this. I don't think there's any _new_ recursion issues that get introduced here and the logic for tracking visited refs is probably sufficient for this scenario?